### ISSUE TITLE
genus return match to species

### ISSUE THREAD

**Issue by BenMerSci** [gnames/gnverifier#125]
When querying through gnverifier for the genus `Oecetis` and authorship `McLachlan, 1877`:
`https://verifier.globalnames.org/api/v1/verifications/Oecetis+McLachlan%2C+1877?capitalize=true&all_matches=true&data_sources=1|3|147`

I believe it gets parsed ok  by gnparser and the result is the same with or without the authorship.

I get two results that are more 'precise' (to the species level with `Oecetis complex`) than what was queried for (to the genus level `Oecetis`).
Is this normal behaviour or a bug? I tried a couple of query for different genus and none return a match to a species level.

**Comment by dimus** [comment]
`Oacetis complex` is returned in your query, because `complex` is a "popular" annotaion word, that parser usually ignores. I have a list of exceptions to the rule, looks like I need to add `Oacetis complex` there

**Comment by dimus** [comment]
hm, it is weird, I do have `Oecetis complex` in exceptions already, I need to investigate. thanks for reporting this @BenMerSci

**Comment by BenMerSci** [comment]
I was just about to reply that I was glad this issue wasn't too much of a hassle because I feel like I'm throwing so many issues at you these days...oops
Let me know if I can do anything!

**Comment by dimus** [comment]
I think I understood what happened. Until last database update I did run global reparsing of all names, however this time, with new system, only new names were parsed, and Oacetis complex is a recently added exception.. So I will need to reparse all name-strings periodically.

You found a real bug, thanks!

**Comment by BenMerSci** [comment]
I'm not sure if this is a parser issue, a CatalogueOfLife issue, or if this is similar to this current issue:

When searching for the genus `Gymnodinium`, CatalogueOfLife returns a match to the species level:
`https://verifier.globalnames.org/api/v1/verifications/Gymnodinium?capitalize=true&all_matches=true&data_sources=1`

which links to this entry:
`https://www.catalogueoflife.org/data/taxon/3HRW2` for the species `Gymnodinium p.dorhni` and authorship `Wawrik, 1956`.

Thoughts?

**Comment by dimus** [comment]
it does look like CoL lost a space between `p.` and `drohni` in this case. I am not sure what `p.` means `Gynodinium p. dorhni` in this case actually, @diatomsrcool, can you help?

**Comment by dimus** [comment]
@BenMerSci in the case of `Gymnodinium p.dorhni` parser correctly considers `p.dorhi` be an unparseable tail, because periods are illegal in specific epithets, and the best result it can produce in this case is genus. So I would say gnverifier behaves as expected.

`Oecetis complex` issue should be fixed now, let me know if you agree, but before closing the ticket lets hear from @diatomsrcool, she is the author of the Gymnodinium checklist.

**Comment by BenMerSci** [comment]
Issue with `Oecetis complex` is indeed fix! Thank you @dimus!

For the `Gymnodinium p.dorhni` issue, the thing is when we query for only the genus `Gymnodinium` like so:
`https://verifier.globalnames.org/api/v1/verifications/Gymnodinium?capitalize=true&all_matches=true&data_sources=1`

we get a result to the species level from CoL (as we can see with the `classificationRanks` key) and the `recordId` key `3HRW2` links to the `Gymnodinium p. dorhni` CoL page `https://www.catalogueoflife.org/data/taxon/3HRW2`.
 But the name of the species returned in the `classificationPath` key is the genus again `Gymnodinium` (probably because the part `p.dorhni` is unparseable?).

**Comment by dimus** [comment]
yes, exactly, the issue is parsing https://parser.globalnames.org/?code=&format=html&names=Gymnodinium+p.+dorhni%0D%0AGymnodinium+p.dorhni&with_details=on

There is not parser rule for names with `p.` coming after genus following specific epithet. If it means `pro parte` it supposed to appear in the end of the name-string.

I will consult ICZN experts about this name.

**Comment by diatomsRcool** [comment]
Sorry! just now seeing this! Yes, Dima is correct.
